### PR TITLE
backport: Merge bitcoin#22496, 21985

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

File: src/netaddress.cpp [hunk 1]
```diff
@@ -542,7 +542,7 @@ static std::string IPv4ToString(Span<const uint8_t> a)
     return strprintf("%u.%u.%u.%u", a[0], a[1], a[2], a[3]);
 }
 
-static std::string IPv6ToString(Span<const uint8_t> a)
+static std::string IPv6ToString(Span<const uint8_t> a, uint32_t scope_id)
```
Comment (UdjinM6) [inline src/netaddress.cpp:545]: 21985: incomplete, must wait for 21756 backport to be merged (via #6008)

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/netaddress.cpp (hunk 1 above)
Comment (knst) [reply src/netaddress.cpp:545]: 6008 is merged

**Review by UdjinM6 — APPROVED** [review]
utACK

NOTE: 22496 is also a part of 6040

**Review by PastaPastaPasta — APPROVED** [review]
utACK ae4c7debc1c7c2d19c087f28a1bc4fe09e34b477

**Pull request merged**